src/main/java/org/jetbrains/bsp/bazel/server/BepServer.java

```diff
@@ -40,30 +35,34 @@
 import java.util.Set;
 import java.util.Stack;
 import java.util.TreeSet;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.bsp.bazel.common.Constants;
 import org.jetbrains.bsp.bazel.common.Uri;
 import org.jetbrains.bsp.bazel.server.logger.BuildClientLogger;
 
 public class BepServer extends PublishBuildEventGrpc.PublishBuildEventImplBase {
 
-  private static final Set<String> SUPPORTED_ACTIONS =
-      ImmutableSet.of(BazelBspServer.KOTLINC, BazelBspServer.JAVAC, BazelBspServer.SCALAC);
-  private final BazelBspServer bspServer;
+  private static final Logger LOGGER = LogManager.getLogger(BepServer.class);
+
   private final BuildClient bspClient;
+  private final BuildClientLogger buildClientLogger;
+
+  private final BazelBspServer bspServer;
+  private final DiagnosticsDispatcher diagnosticsDispatcher;
+
+  private final Stack<TaskId> taskParkingLot = new Stack<>();
```

Comment:
I renamed it and changed it to Deque because Stack is legacy, but what exactly is your idea with the variables? The only way I see it to have two attributes: startedEventTaskId and _uniqueStartedEvent_ and then having something very ugly like:

```
if (startedEventTaskId == null) {
startedEventTaskId = taskId;
} else {
_uniqueStartedEvent_ = true;
}
```

(or the same thing with some Optional but it doesn't look better imo)